### PR TITLE
Collapsible Platform Edges.

### PR DESCRIPTION
**Pull request by Venuska1117**
**!!!DISCLAIMER!!!
I am very bad at spriting and i am not joking.**

**_I recommend to put this PR in Testmerge, so i can find issues caused by swapping paths and  find oversights with sprites._**

# About the pull request
I am working on rework of Kutjevo #6971

Fixes: #7049

Because i plan to add elevation by using large quantities of platform edges, i tough it will be too annoying for players, who are keep dying, like xenos and marines because edge stopped them for 1-2 seconds.

**Edit: Its 70% done, i just lack sprites. _(made only one platform sprite set for demonstration in videos below)_**

Edit(07.09): I was forced to change paths of platforms so they can play different sounds depending on what they are made of.

Edit(15.09): Finally done sprites, now time to wait.

Edit(21.09): thanks to **Doubleumc | Casey Walton** and **cuberound** for having time explain me some code related terms, thanks you both for that.

Edit(15.10): Added Stair variant of Platform with own sprites and broken states.

# Explain why it's good for the game

Allow game to be more dynamic by adding little more "terrain destruction" that will benefit gameplay and mappers who like to keep Map elevation but they are worried that platform edges will make experience negative. (people complain a lot)

# Testing Photographs and Procedure

<details>
<summary>Click HERE to see Videos</summary>

Platforms **before** destruction:
New1

Platforms **after** destruction:
New2

How it work and sound when xeno break them:

https://github.com/user-attachments/assets/de9e8c9e-ffaf-4544-beab-a58640b857fa

How it will work for Explosions:

https://github.com/user-attachments/assets/cbc5e4bb-1603-42bf-aa02-c3e875f86292

</details>

# Changelog

:cl: Venuska1117
code: Added ability to Destroy Platform Edges by xenos on "Disarm" intent, now depending on xenos "size" they can break Platform Edges quicker or Slower, Now Explosives have chance to destroy Platform Edges by depending on how strong explosion is.
imageadd: Added Sprites for Broken Platforms Edges.
soundadd: Added Sound for Breaking "Stone" Edges.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/structures/platforms.dm
Comment (kiVts) [inline code/game/objects/structures/platforms.dm:66]:
```suggestion
		. += SPAN_WARNING("It looks destroyed.")
```

File: code/game/objects/structures/platforms.dm
Comment (kiVts) [inline code/game/objects/structures/platforms.dm:126]: if you want for all xeno to have diffrent time to break the ledge, add a var on xeno type, an easier approach would be to look at the mob size because you dont need to add var at xeno type. But this is disgusting.

File: code/game/objects/structures/platforms.dm
Comment (kiVts) [inline code/game/objects/structures/platforms.dm:130]: early return

**Review by kiVts — CHANGES_REQUESTED** [review]

**Review by kiVts — DISMISSED** [review]

File: maps/map_files/DesertDam/sprinkles/10.damtemple_intact.dmm
Comment (Nanu308) [inline maps/map_files/DesertDam/sprinkles/10.damtemple_intact.dmm:1]: Check the corners for the platforms, issue with other .dmm files as well.

**Review by Nanu308 — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Nanu308 — CHANGES_REQUESTED** [review]
Use the mapdiffbot, double check ingame after Initialize has happened.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by Nanu308** [comment]
Mapping lgtm, need @Blundir approval for sprites and then a code check

**Review by Nanu308 — CHANGES_REQUESTED** [review]
nvrmind, see comment in Discord related to South facing and linking

**Review by Blundir — APPROVED** [review]
sprites seem fine, no problem with the sprite reorg

**Review by Drulikar — CHANGES_REQUESTED** [review]
- Solars on all maps: the south facing sprites seem to have changed for the worse
- Hunter ship has changed heavily for the worse
- Some layering changes like the cliffs on Soro are more misleading now.

**Review by Nanu308 — APPROVED** [review]
Mapping now looks good on mapdiff bot, checked every instance, Needs code approval from @Drulikar

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:109]: Why is this here? You aren't opening a TGUI interface to break walls.
```suggestion
```

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:107]: Move the `action_busy` and `user.a_intent != INTENT_DISARM` check before the `BROKEN` and `explo_proof` checks because otherwise you get error messages when you aren't even in disarm mode clicking on platforms.

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:130]: Since this can be a long do_after, you need to check more often for whether an interrupt has occurred. 
```suggestion
	if(!do_after(user, shove_time, INTERRUPT_ALL, BUSY_ICON_HOSTILE, numticks = shove_time * 0.1))
```

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:110]: Are you sure you want a attack delay for doing nothing?
```suggestion
		return XENO_NO_DELAY_ACTION
```

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:185]:
```suggestion
	dir = NORTH
```

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:217]:
```suggestion
	dir = NORTH
```

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:64]:
```suggestion
```

File: code/game/objects/structures/platforms.dm
Comment (Drulikar) [inline code/game/objects/structures/platforms.dm:53]:
```suggestion
			if(prob(15))
				broken()
				return
		if(EXPLOSION_THRESHOLD_LOW to EXPLOSION_THRESHOLD_HIGH)
			if(prob(30))
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Review by Drulikar — APPROVED** [review]

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Pull request merged**
